mlir/include/mlir/Dialect/Rock/utility/transformMapUtils.h

```diff
@@ -61,6 +61,14 @@ bool needs64BitIndices(TransformMapAttr map);
 /// Apply a chain of transforms on a memref and return the final view
 Value transform(OpBuilder &b, Value toBeTransformed, ArrayAttr transforms);
 
+/// Returns a version of `transformed` where all the `rock.transform`
+/// and `rock.scalarize` operations between `transformed` and the underlying
+/// memory have one user, cloning those intermediate operations if needed.
+/// This ensures that optimizations can edit those transformations in place
+/// without breaking any other uses that may have been merged together.
+/// If the transforms are already isolated, this function does nothing.
+Value isolateTransforms(OpBuilder &b, Value transformed);
```

Comment:
Generally I feel this is a useful canonicalization for rock.transform ops because it is always going to be replicated at the ultimate user. WDYT ? 

No need to do anything here; just a comment.

Reply:
On the other hand, transforms are `Pure`, should die when they have no users, etc., so this is temporarily breaking common subexpression elimination.

Though I'd at least consider whether `rock.transform` could take an array of transforms ...